### PR TITLE
Error in push finish

### PR DESCRIPTION
**Pull request by harminius**
Task: https://github.com/MerginMaps/server-private/issues/2558

Situation:
When migrating project files project without a file were given Null value to the file history id. We expect this field not be Null. The caching method which should update the field fails to do it because it cannot find a file associated with the project id.

Solution
When no files are found when caching the latest files, put an empty array in the column and don't keep it Null.

Bonus: 
Fix flask cmd to create a project.

### CODE REVIEW

File: server/mergin/sync/models.py [hunk 1]
```diff
@@ -743,7 +743,9 @@ def __init__(
         latest_files_map = {
             fh.path: fh.id
             for fh in FileHistory.query.filter(
-                FileHistory.id.in_(self.project.latest_project_files.file_history_ids)
+                FileHistory.id.in_(
+                    self.project.latest_project_files.file_history_ids or {}
```
Comment (varmar05) [inline server/mergin/sync/models.py:747]: Could you add a test?

**Review by varmar05 — CHANGES_REQUESTED** [review]

File: server/mergin/sync/commands.py [hunk 2]
```diff
@@ -28,6 +29,7 @@ def project():
     def create(name, namespace, user):  # pylint: disable=W0612
         """Create blank project"""
         workspace = current_app.ws_handler.get_by_name(namespace)
+        user = User.query.get(int(user))
```
Comment (varmar05) [inline server/mergin/sync/commands.py:32]: I would probably go with username

**Comment by varmar05** [comment]
"Make file_history_ids not nullable?" It was meant to be nullable and updated only on demand (first pull / get info) as whole migration would have been very heavy. So probably we maybe want to cache files ids if it is null as a fix.

File: server/mergin/sync/commands.py (hunk 2 above)
Comment (harminius) [reply server/mergin/sync/commands.py]: Done

File: server/mergin/sync/models.py (hunk 1 above)
Comment (harminius) [reply server/mergin/sync/models.py]: Done

**Comment by harminius** [comment]
> "Make file_history_ids not nullable?" It was meant to be nullable and updated only on demand (first pull / get info) as whole migration would have been very heavy. So probably we maybe want to cache files ids if it is null as a fix.

Actually, I found out the latest project files are cached in this scenario.

That's why the problem is only in empty projects. The caching method finds no files to cache and keeps the column `null`.

**Comment by varmar05** [comment]
> Actually, I found out the latest project files are cached in this scenario.
> 
> That's why the problem is only in empty projects. The caching method finds no files to cache and keeps the column `null`.

If caching method finds no files then I guess it should set it to empty array rather that NULL which should be only used if that row has never been cached.

**Pull request merged**